evap/evaluation/models.py

```diff
@@ -1044,6 +1044,16 @@ def unlogged_fields(self):
     def is_general(self):
         return self.contributor_id is None
 
+    @property
+    def can_be_deleted(self):
+        if self.pk is None:
+            return True  # not stored in the DB. Required so temporary instances in the formset can be deleted.
+
+        if not self.ratinganswercounter_set.exists() and not self.textanswer_set.exists():
```

Comment:
It seems odd that the knowledge about the possible kinds of answers is also kept here, would there be any way to avoid that (without loosing the advantages @He3lixxx named)? Or should we add an assertion about the available subclasses like on line 462 in `student/*/test_views.py` to the new test?

Reply:
Adding an assertion either to the test or to the method is probably a good idea.

I'd be slightly in favor of adding it to the method because that's where the assertion is really needed, and the test would fail in both ways, but considering that we added it to the test in #1549, that's probably what we should do for uniformity.

Reply:
Seems reasonable, I just copied the assertion from #1549 :P